Doc additional tf 012 sentinel diffs

rberlind opened the pull request:
This PR points out some additional important differences in Sentinel between Terraform 0.11 and 0.12.

I added additional differences between how Sentinel behaves in Terraform 0.11 and 0.12
Near the beginning of this document, I mention that numeric attributes are now treated as floats.
Also, since we now have the tfrun import (which did not exist when this document was first written), I added a paragraph about that, saying that there are no cost_estimate data in Terraform 0.11.
I also described the diffences when resources are destroyed but not re-created, referring to the new destroy and requires_new attributes.

I also corrected a typo "cost a estimate" to "a cost estimate" in the tfrun doc and mentioned that cost estimates are never available in Terraform 0.11

vancluever commented on content/source/docs/cloud/sentinel/sentinel-tf-012.html.md:
```diff
@@ -170,6 +184,36 @@ namespace](./import/tfplan.html#namespace-resource-diff)
+
+In Terraform 0.11, when a resource is being destroyed but not re-created, it's
+[`diff`](./import/tfplan.html#value-diff) value in the `tfplan` import is empty
+while its [`applied`](./import/tfplan.html#value-applied) value has data. In
```
We have tests indicating that `applied` should be empty even on TF 0.11. Let's discuss this offline.

rberlind replied:
As we discussed offline, applied no longer shows up during destroy, even in TF 0.11 due to a change in a PR in the sentinel-plugin-terraform rep. I will remove the reference to that.

vancluever commented on content/source/docs/cloud/sentinel/sentinel-tf-012.html.md:
```diff
@@ -170,6 +184,36 @@ namespace](./import/tfplan.html#namespace-resource-diff)
+see if a resource is being destroyed and not re-created with both versions of
+Terraform.
+
+Please note that if you are using a private instance of Terraform Enterprise,
```
Let's remove the extra "private" stuff here to be in line with the new branding.

```suggestion
Please note that if you are using Terraform Enterprise,
```

vancluever commented on content/source/docs/cloud/sentinel/sentinel-tf-012.html.md:
```diff
@@ -170,6 +184,36 @@ namespace](./import/tfplan.html#namespace-resource-diff)
+Fortunately, after Terraform 0.12 was released, new
+[`destroy`](./import/tfplan.html#value-destroy) and
+[`requires_new`](./import/tfplan.html#value-requires_new) values were added to
+the `tfplan` import.  Since these values are available with both Terraform
```
```suggestion
the `tfplan` import. Since these values are available with both Terraform
```

rberlind commented:
I commited both of your suggestions, @vancluever, and also addressed your comment about applied not being present for destroyed resources in both Terraform 0.11 and 0.12.  I also streamlined some of the text.

glenngillen commented on content/source/docs/cloud/sentinel/import/tfrun.html.md:
```diff
@@ -118,7 +118,8 @@ vcs_repo (map of keys)
 
 The **cost_estimation namespace** contains data associated with the current run's cost estimate. This
-namespace is only present if cost a estimate is available.
+namespace is only present if a cost estimate is available. Note that cost estimates are never
```
"never" feels a bit strong here. Could we switch it to just say "not"?

rberlind replied:
Sure

glenngillen commented on content/source/docs/cloud/sentinel/sentinel-tf-012.html.md:
```diff
@@ -17,9 +17,20 @@ modification.
+checks and comparisons done against numeric attributes.
+
+There are no explicit changes in the `tfrun` import for Terrraform 0.12, but
+the [`cost_estimate namespace`](./import/tfrun.html#namespace-cost_estimate)
```
Should it be:

- `cost_estimate namespace`, or
- `cost_estimate` namespace

I'm low confidence on what the right answer is, just jumped out at me and thought I'd flag it for consideration. Happy to defer judgement to the experts.

rberlind replied:
I believe `cost_estimate namespace` was already used.  And the doc definitely did already have `workspace namespace`.  So, what I did is consistent.

vancluever replied:
I did take a look at this and I think that the issue that Glenn was mentioning here is correct in that the formatting should be "`cost_estimate` namespace". The other one reads "resource namespace" without pre-formatting so it's not _exactly_ the same thing.
 
I fixed this though so it doesn't have to be anymore, this is good to merge now without any more changes.

glenngillen commented on content/source/docs/cloud/sentinel/sentinel-tf-012.html.md:
```diff
@@ -170,6 +184,35 @@ namespace](./import/tfplan.html#namespace-resource-diff)
+
+In Terraform 0.11, when a resource is being destroyed but not re-created, it's
+[`diff`](./import/tfplan.html#value-diff) value in the `tfplan` import is empty.
+In Terraform 0.12, however, the `diff` value does has data. For both versions of
```
"value does ~has~have data." or "value ~does~ has data."

rberlind replied:
thanks for catching this.  Fixed.

glenngillen commented on content/source/docs/cloud/sentinel/sentinel-tf-012.html.md:
```diff
@@ -170,6 +184,35 @@ namespace](./import/tfplan.html#namespace-resource-diff)
+
+Please note that if you are using Terraform Enterprise,
+you must use version v201909-1 or higher in order to use the `destroy` and
+`requires_new` values.
```
Suggested edit:

> absent. This also means previous policies that tested whether a resource was being destroyed and not re-created by testing the condition `length(r.diff) == 0` will need to be updated.
>
>New [`destroy`](./import/tfplan.html#value-destroy) and [`requires_new`](./import/tfplan.html#value-requires_new) values have been added to the `tfplan` import. Since these values are available both for Terraform 0.11 and 0.12, you can now test `r.destroy and not r.requires_new` to see if a resource is being destroyed and not re-created with both versions of Terraform.
>
>It is therefore recommended to update all policies, in both Terraform 0.11 and 0.12, that use the `tfplan` import and the `applied` value to check whether resources are being destroyed and not re-created. This will avoid `undefined` values in your functions and rules.
>
>Please note that if you are using Terraform Enterprise, you must use version v201909-1 or higher in order to use the `destroy` and `requires_new` values.

rberlind replied:
Thanks @glenngillen .  I revised the passage to include your suggestions, although I did not word things exactly as you did.  A new thing I mentioned which I learned about yesterday in discussions with @vancluever is that we recently made a change to the `tfplan` import so that the `applied` value is not present when a resource is being destroyed but not re-created.  Since that can affect existing policies, I thought it important to point that out.  I also made the doc's use of ticks more consistent and removed some excessive hyperlinks.

rberlind commented:
Thanks @vancluever
